cakeshop-integration-test/src/test/java/uk/gov/justice/services/cakeshop/it/RebuildSnapshotsIT.java

```diff
@@ -0,0 +1,227 @@
+
+        batchEventInserter.updateEventLogTable(events);
+
+        final JmxCommandRuntimeParameters aggregateClassNameAndStreamId = new JmxCommandRuntimeParameters.JmxCommandRuntimeParametersBuilder()
+                .withCommandRuntimeString(aggregateClassName)
+                .withCommandRuntimeId(recipeId)
+                .build();
+
+        try (final SystemCommanderClient systemCommanderClient = systemCommanderClientFactory.create(JmxParametersFactory.buildJmxParameters())) {
+            final SystemCommanderMBean systemCommanderMBean = systemCommanderClient.getRemote(CONTEXT_NAME);
+            final UUID commandId = systemCommanderMBean.call(
+                    REBUILD_SNAPSHOTS,
+                    aggregateClassNameAndStreamId,
+                    FORCED);
+
+            final StopWatch stopWatch = new StopWatchFactory().createStartedStopWatch();
+            final Optional<SystemCommandStatus> systemCommandStatus = longPoller.pollUntilFound(() -> {
+                final SystemCommandStatus commandStatus = systemCommanderMBean.getCommandStatus(commandId);
+                final String systemCommandName = commandStatus.getSystemCommandName();
+                final CommandState commandState = commandStatus.getCommandState();
+
+                System.out.println("JMX command '" + systemCommandName + "' state is '" + commandStatus.getCommandState() + "'");
+
+                if (commandState.equals(COMMAND_COMPLETE)) {
+                    return Optional.of(commandStatus);
+                }
+
+                return empty();
+            });
+
+            stopWatch.stop();
+
+            if (systemCommandStatus.isPresent()) {
+                assertThat(systemCommandStatus.get().getCommandState(), is(COMMAND_COMPLETE));
+                final Optional<AggregateSnapshot<Recipe>> latestSnapshot = snapshotJdbcRepository.getLatestSnapshot(
+                        recipeId,
+                        Recipe.class);
+
+                assertThat(latestSnapshot.isPresent(), is(true));
+
+                System.out.println("Run of JMX command '" + REBUILD_SNAPSHOTS + "' took " + stopWatch.getTime() + " milliseconds");
+
+            } else {
+                fail();
+            }
+        }
+    }
+
+    private void cleanViewstoreTables() {
+        databaseCleaner.cleanViewStoreTables(DB_CONTEXT_NAME,
+                "ingredient",
+                "recipe",
+                "cake",
+                "cake_order",
+                "processed_event"
+        );
+        databaseCleaner.cleanStreamBufferTable(DB_CONTEXT_NAME);
+        databaseCleaner.cleanStreamStatusTable(DB_CONTEXT_NAME);
+    }
+
+    private List<Event> createEvents(final long numberToCreate, final UUID streamId) {
+
+        final long firstPositionInStream = 1;
+        final ArrayList<Event> events = new ArrayList<>();
+        final Event recipeAddedEvent = createRecipeAddedEvent(firstPositionInStream, streamId);
+        events.add(recipeAddedEvent);
```

Comment:
we could have the loop below start with 1 and have this code moved into the loop?

Reply:
They are different events :-( so I don't think that's possible. 
The first call creates a 'cakeshop.events.recipe-added' event, the subsequent calls create 'cakeshop.events.recipe-renamed' events